solidity/app/test/router.test.ts

```diff
@@ -84,25 +100,128 @@ describe('Router', async () => {
     ).to.be.revertedWith(ONLY_OWNER_REVERT_MSG);
   });
 
-  it('dispatches message to enrolled remote router', async () => {
-    const outboxFactory = new Outbox__factory(signer);
-    const outbox = await outboxFactory.deploy(origin);
-    await connectionManager.setOutbox(outbox.address);
-
-    const remote = nonOwner.address;
-    await router.enrollRemoteRouter(
-      destination,
-      utils.addressToBytes32(remote),
-    );
-    await expect(router.dispatchToRemoteRouter(destination, message)).to.emit(
-      outbox,
-      'Dispatch',
-    );
+  describe('dispatch functions', () => {
+    let interchainGasPaymaster: InterchainGasPaymaster;
+    beforeEach(async () => {
+      const interchainGasPaymasterFactory = new InterchainGasPaymaster__factory(
+        signer,
+      );
+      interchainGasPaymaster = await interchainGasPaymasterFactory.deploy();
+      await connectionManager.setInterchainGasPaymaster(
+        interchainGasPaymaster.address,
+      );
+
+      // Enroll a remote router on the destination domain.
+      // The address is arbitrary because no messages will actually be processed.
+      await router.enrollRemoteRouter(
+        destination,
+        utils.addressToBytes32(nonOwner.address),
+      );
+    });
+
+    // Helper for testing different variations of dispatch functions
+    const runDispatchFunctionTests = async (
+      dispatchFunction: (
+        destination: number,
```

Comment:
This is the destination domain right? We should probably be more consistent across, as the parameter is called `destinationDomain` in `Outbox.sol`